marlowe-playground-client/src/MarloweEditor/Types.purs

```diff
@@ -14,12 +14,42 @@ import Data.Maybe (Maybe(..))
 import Data.Symbol (SProxy(..))
 import Halogen.Monaco (KeyBindings(..))
 import Halogen.Monaco as Monaco
-import Marlowe.Extended (IntegerTemplateType)
+import Marlowe.Extended (ContractType, IntegerTemplateType, MetadataHintInfo)
+import Marlowe.Semantics as S
 import Monaco (IMarkerData)
 import StaticAnalysis.Types (AnalysisState, initAnalysisState)
 import Text.Parsing.StringParser (Pos)
 import Web.HTML.Event.DragEvent (DragEvent)
 
+class ShowConstructor a where
+  showConstructor :: a -> String
+
+data MetadataAction
+  = SetContractName String
+  | SetContractType ContractType
+  | SetContractDescription String
+  | SetRoleDescription S.TokenName String
+  | DeleteRoleDescription S.TokenName
+  | SetSlotParameterDescription String String
+  | DeleteSlotParameterDescription String
+  | SetValueParameterDescription String String
+  | DeleteValueParameterDescription String
+  | SetChoiceDescription String String
+  | DeleteChoiceDescription String
```

Comment:
I think we can make do without the various `Delete` actions, can't we? Why would you want to delete a description? Also, see the comments below about initialising the metadata with empty descriptions for everything.

Reply:
Well I was thinking we would delete them automatically. We would just constantly keep the metadata maps in sync with the template. I think I still lean slightly towards doing things this way, but... Your call. 🤷‍♂️